components/editor/modules/meta/ShareImageForm.js

```diff
@@ -0,0 +1,196 @@
+  ])
+  const initData = initValues.merge(
+    data.filter((_, key) => initValues.has(key))
+  )
+  return (
+    <MetaForm
+      data={initData}
+      onInputChange={onInputChange}
+      getWidth={() => 620}
+    />
+  )
+}
+
+const UploadImage = withT(({ t, data, onInputChange, socialKey }) => {
+  const imageKey = socialKey + 'Image'
+  const labelHeight = 17 + 5
+
+  return (
+    <div style={{ height: 314 + labelHeight, width: 600, overflow: 'hidden' }}>
+      <ImageInput
+        maxWidth='100%'
+        maxHeight={314}
+        imageStyles={socialPreviewStyles[socialKey]}
+        label={t(`metaData/field/${imageKey}`)}
+        src={data.get(imageKey)}
+        onChange={onInputChange(imageKey)}
+      />
+    </div>
+  )
+})
+
+const GenerateImage = ({ data, onInputChange, format }) => (
+  <ShareImageGenerator
+    format={format}
+    fontSize={data.get('shareFontSize')}
+    onFontSizeChange={(e, value) =>
+      onInputChange('shareFontSize')(undefined, Number(value) || undefined)
+    }
+    onFontSizeInc={() =>
+      onInputChange('shareFontSize')(
+        undefined,
+        (data.get('shareFontSize') || 60) + 1
+      )
+    }
+    onFontSizeDec={() =>
+      onInputChange('shareFontSize')(
+        undefined,
+        Math.max(data.get('shareFontSize') - 1, 0)
+      )
+    }
+    inverted={data.get('shareInverted')}
+    onInvertedChange={onInputChange('shareInverted')}
+    text={data.get('shareText')}
+    onTextChange={e => onInputChange('shareText')(undefined, e.target.value)}
+    textPosition={data.get('shareTextPosition')}
+    onTextPositionChange={item =>
+      onInputChange('shareTextPosition')(undefined, item.value)
+    }
+  />
+)
+
+const PreviewText = ({ data, socialKey }) => {
+  const Preview = previews[socialKey]
+  return (
+    <Preview
+      title={data.get(socialKey + 'Title') || data.get('title')}
+      description={
+        data.get(socialKey + 'Description') || data.get('description')
+      }
+    />
+  )
+}
+
+const ShareImageForm = withT(({ t, editor, node, onInputChange, format }) => {
+  const [generated, setGenerated] = useState(
+    !!format || !!node.data.get('shareText')
+  )
+
+  useEffect(() => {
+    if (generated) {
+      editor.change(change => {
+        change.setNodeByKey(node.key, {
+          data: node.data
+            .set(
+              'shareTextPosition',
+              node.data.get('shareTextPosition') ||
+                SHARE_IMAGE_DEFAULTS.textPosition
+            )
+            .set(
+              'shareFontSize',
+              node.data.get('shareFontSize') || SHARE_IMAGE_DEFAULTS.fontSize
+            )
+        })
+      })
+    } else {
+      editor.change(change => {
+        change.setNodeByKey(node.key, {
+          data: node.data
+            .remove('shareInverted')
+            .remove('shareTextPosition')
+            .remove('shareFontSize')
+            .remove('shareText')
+        })
+      })
+    }
+  }, [generated])
```

Comment:
```suggestion
  const generated = !!(node.data.get('shareText') || node.data.get('shareFontSize') || node.data.get('shareTextPosition'))

  const setGenerated = (generated) => {
    if (generated) {
      editor.change(change => {
        change.setNodeByKey(node.key, {
          data: node.data
            .set(
              'shareTextPosition',
              node.data.get('shareTextPosition') ||
                SHARE_IMAGE_DEFAULTS.textPosition
            )
            .set(
              'shareFontSize',
              node.data.get('shareFontSize') || SHARE_IMAGE_DEFAULTS.fontSize
            )
        })
      })
    } else {
      editor.change(change => {
        change.setNodeByKey(node.key, {
          data: node.data
            .remove('shareInverted')
            .remove('shareTextPosition')
            .remove('shareFontSize')
            .remove('shareText')
        })
      })
    }
  }
```

Reply:
Ufff, nobody has a right to be so clever, nobody!